### DISCUSSION TITLE
padding between series in stacked bar chart

### DISCUSSION THREAD

**Discussion by brownjedi** [airbnb/visx#1459] [Q&A]
Hi,

Thank you such an amazing visualization library. I'm having a blast using this. 

I have a quick question, is it possible to do a padding between series in a Bar Stack XY chart?

Cheers,
Sai

![Screen Shot 2022-03-30 at 9 28 39 AM](https://user-images.githubusercontent.com/6647593/160845988-45f66297-2edd-4de5-95ef-876e95e2a6bd.png)

**Comment by williaster** [comment]
hey @brownjedi 👋  thanks for checking out visx, glad you are enjoying it. bar padding in `@visx/xychart` is controlled by the scales, this concept is inherited from `d3`: https://github.com/d3/d3-scale#band-scales

<img width="800" src="https://user-images.githubusercontent.com/4496521/105773251-77481a80-5f18-11eb-9727-25749763da78.png" />

so overall you'd set some combination of `padding/paddingInner/paddingOuter` on your `xScale` config and that should give you the results you want. here's a [sandbox](https://codesandbox.io/s/bar-padding-demo-z52023?file=/Example.tsx) to tinker with.

```tsx
<XYChart
  xScale={{
    type: "band",
    paddingInner: 0.5
    // paddingOuter: 5
    // padding: 0, // use this to set inner + outer to the same value
  }}
  {...}
>
```

![image](https://user-images.githubusercontent.com/4496521/160921414-de3b6534-92e2-4ec7-a930-d184b979e660.png)

**Reply by brownjedi** [reply]
hey @williaster 

Thank you for the quick response. I was thinking about the vertical padding highlighted here. I was thinking one option would be to introduce a fake series and set the color white or transparent to get the desired effect, but was wondering if there is a better way to do it.

![Screen Shot 2022-03-31 at 10 38 23 AM](https://user-images.githubusercontent.com/6647593/161081689-ed472b14-f031-4f28-8852-e3ea3c7aff58.png)
